src/CodeGen/ODataT4CodeGenerator.ttinclude

```diff
@@ -317,6 +327,50 @@ public void ValidateAndSetIgnoreUnexpectedElementsAndAttributesFromString(string
     this.IgnoreUnexpectedElementsAndAttributes = boolValue;
 }
 
+/// <summary>
+/// Set the CustomHttpHeaders property from the custom http headers string.
+/// </summary>
+/// <param name="stringValue">The value to set.</param>
+public void ValidateAndSetCustomHttpHeadersFromString(string headersValue) 
+{
+    if (String.IsNullOrWhiteSpace(headersValue))
+    {
+        return;
+    }
+
+    this.CustomHttpHeaders = new List<string>();
+
+    string[] headerElements = headersValue.Split(new string[] { Environment.NewLine },StringSplitOptions.None);
+    foreach (var headerElement in headerElements)
+    {
+        // Trim header for empty spaces
+        var header = headerElement.Trim();
+
+        // A header string must have atleast one semi-colon
+        if (GetSemiColons(header) > 0)
+        {
+            CustomHttpHeaders.Add(header);
+        }
+        else
+        {
+            throw new ArgumentException("A http header string must have a semi-colon delimeter");
+        }
+    }
+
+}
+
+private static int GetSemiColons(string input)
+{
+    // Remove white spaces
+    string headers = Regex.Replace(input, @"\s", "");
+
+    //Count matched closing parantheses
+    int matchedSemiColons = Regex.Matches(headers, @"\:").Count;
+
+    // If valid, delimeters = closing parantheses - 1
+    return matchedSemiColons;
+}
+

```

Comment:
Check the comments left on ODataT4CodeGenerator.cs file

Reply:
Fixed